### PR TITLE
Test net changes

### PR DESCRIPTION
**Pull request by Andrew-Pohl**
Changes to allow spark to be maintained alongside mainnet

### CODE REVIEW

File: Version_testNet [hunk 1]
```diff
@@ -0,0 +1,3 @@
+DOCKER_IMAGE_FUSE_APP_VERSION="1.0.0"
```
Comment (leonprou) [inline Version_testNet:1]: Can you add an empty line for all the files where it's missing? @Andrew-Pohl

File: Dockerfile.spark [hunk 2]
```diff
@@ -0,0 +1,32 @@
+FROM ubuntu:18.04
+
+ENV HOME=/home/parity
```
Comment (leonprou) [inline Dockerfile.spark:3]: Dockerfile.spark sounds cool

File: config/spark/spec.json [hunk 3]
```diff
@@ -0,0 +1,98 @@
+{
+	"name": "FuseNetwork",
+	"engine": {
+		"authorityRound": {
+			"params": {
+				"stepDuration": "5",
```
Comment (leonprou) [inline config/spark/spec.json:6]: That's the contract you deployed, right?
What's the cycle length for Consensus?

File: scripts/examples/.env.bootnode.example [hunk 4]
```diff
@@ -5,3 +5,6 @@ BOOTNODES=enode://e21e2053005e40653d055fc01a07768357749d27d630702c203b1a3c00bdf2
 
 # in case `sudo` is needed
 PERMISSION_PREFIX=
+
+# Set to true to run on spark testnet
+TESTNET=false
```
Comment (leonprou) [inline scripts/examples/.env.bootnode.example:10]: I wonder if we need this as that's the default..
I prefer to hid that config from validators. So they don't do `TESTNET=true` for some reason

File: scripts/quickstart.sh [hunk 5]
```diff
@@ -290,6 +292,16 @@ function parseArguments {
...
     fi
   fi
+  
+  if [ -z "$TESTNET" ] ; then
+    if [ $TESTNET == "true" ]
+      if [[ $ROLE == bridge-validator ]] ; then
+        displayErrorAndExit "bridge-validators not supported on Spark"
+      fi
+      TEST_NET_STRING = "-spark"
+      VERSION_FILE = "https://raw.githubusercontent.com/fuseio/fuse-network/master/Version_testNet"
```
Comment (leonprou) [inline scripts/quickstart.sh:302]: can you rename the version file to `Version_spark-testnet` to be aligned with test of the naming

File: scripts/quickstart.sh [hunk 6]
```diff
@@ -361,9 +373,9 @@ function setup {
 
   # Pull the docker images.
   echo -e "\nPull the docker images..."
-  DOCKER_IMAGE_PARITY="$DOCKER_IMAGE_PARITY:$DOCKER_IMAGE_FUSE_PARITY_VERSION"
-  DOCKER_IMAGE_NETSTAT="$DOCKER_IMAGE_NETSTAT:$DOCKER_IMAGE_NET_STATS_VERSION"
-  DOCKER_IMAGE_APP="$DOCKER_IMAGE_APP:$DOCKER_IMAGE_FUSE_APP_VERSION"
+  DOCKER_IMAGE_PARITY="$DOCKER_IMAGE_PARITY$TEST_NET_STRING:$DOCKER_IMAGE_FUSE_PARITY_VERSION"
```
Comment (leonprou) [inline scripts/quickstart.sh:376]: Can this be problematic that both main and testnet will use the same docker versions for the containers `DOCKER_IMAGE_FUSE_PARITY_VERSION` ?

File: scripts/quickstart.sh [hunk 7]
```diff
@@ -361,9 +373,9 @@ function setup {
...
+  DOCKER_IMAGE_NETSTAT="$DOCKER_IMAGE_NETSTAT$TEST_NET_STRING:$DOCKER_IMAGE_NET_STATS_VERSION"
+  DOCKER_IMAGE_APP="$DOCKER_IMAGE_APP$TEST_NET_STRING:$DOCKER_IMAGE_FUSE_APP_VERSION"
```
Comment (leonprou) [inline scripts/quickstart.sh:378]: If you rename this to `TEST_NET_SUFFIX` I think it'll be more clear why we add this here

**Review by leonprou — COMMENTED** [review]
Looks good.
Wrote some comments and question. But I think we can start with deploying this.
Need to discuss docker repo for this. I think we can use a separate repo

File: Version_testNet (hunk 1 above)
Comment (Andrew-Pohl) [reply Version_testNet:1]: Not sure I get you here like a change in EOL characters (should be unix LF), or do you mean a trailing new line at EOF?

File: config/spark/spec.json (hunk 3 above)
Comment (Andrew-Pohl) [reply config/spark/spec.json:6]: No not yet haven't deployed so the address are subject to change :)

File: scripts/examples/.env.bootnode.example (hunk 4 above)
Comment (Andrew-Pohl) [reply scripts/examples/.env.bootnode.example:10]: Yes might be a good idea to avoid confusion

File: scripts/quickstart.sh (hunk 6 above)
Comment (Andrew-Pohl) [reply scripts/quickstart.sh:376]: They will use a different version file, the version file gets overwritten on line 302 if running on testnet

**Comment by leonprou** [comment]
@Andrew-Pohl You need to push he latest changes here, right?

**Comment by Andrew-Pohl** [comment]
> @Andrew-Pohl You need to push he latest changes here, right?

Yep need to push the docker containers and update the qs to point to them 👍

**Comment by Andrew-Pohl** [comment]
@leonprou good to go now all images have been uploaded and the network is ticking along https://health.fusespark.io 

Will get the explorer up this week

**Review by leonprou — APPROVED** [review]

**Review by leonprou — APPROVED** [review]

**Pull request merged**